packages/valory/skills/decision_maker_abci/behaviours/decision_receive.py

```diff
@@ -505,9 +563,16 @@ def async_act(self) -> Generator:
                     prediction_response,
                     bet_amount,
                 )
-                next_mock_data_row = self.synchronized_data.next_mock_data_row + 1
-
-            self._update_selected_bet()
+            # always remove the processed trade from the benchmarking input file
+            # now there is one reader pointer per market
+            if self.benchmarking_mode.enabled:
+                # always remove the processed trade from the benchmarking input file
+                # now there is one reader pointer per market
+                bet = self.get_active_sampled_bet()
+                if self.shared_state.bet_id_row_manager[bet.id]:
+                    self.shared_state.bet_id_row_manager[bet.id].pop(0)
```

Comment:
Assing `self.shared_state.bet_id_row_manager[bet.id]` to some var first so that it's more readable.

Reply:
done.